src/utilities/Remotes.jl

```diff
@@ -103,7 +103,30 @@ function repofile(remote::Remote, ref, filename, linerange = nothing)
 end
 
 """
-    GitHub(user :: AbstractString, repo :: AbstractString)
+    parse_url(url)
+
+Return named tuple with values `scheme`, `authority`, `userinfo`, `username`, `password`, `hostname`, `port`, `path`, `query`, `fragment`.
+If certain value is not present in URL then it is omitted in return tuple.
+"""
+function parse_url(url)
+    m = match(r"^(?:(?P<scheme>https?\:)\/\/)?(?P<authority>(?:(?:(?P<userinfo>(?P<username>[^:]+)(?::(?P<password>[^@]+)?)?)@)?)(?P<hostname>[^:\/?#]*)(?:\:(?P<port>[0-9]+))?)(?P<path>[\/]{0,1}[^?#]*)(?P<query>\?[^#]*|)(?P<fragment>#.*|)$", url)
+    return NamedTuple((name => m[name] for name in (:scheme, :authority, :userinfo, :username, :password, :hostname, :port, :path, :query, :fragment)))
+end
+
+"""
+    github_host()
+
+Returns hostname of the GitHub installation it is running on.
+Is determined by `ENV[GITHUB_SERVER_URL]` variable which is set by GitHub Actions runtime.
```

Comment:
Some grammar issues; maybe this is a bit better?
```suggestion
This is derived from the `ENV[GITHUB_SERVER_URL]` variable which is set in every GitHub Actions workflow.
```